controller/sync.go

```diff
@@ -152,7 +152,8 @@ func (m *appStateManager) SyncAppState(app *v1alpha1.Application, state *v1alpha
 		revisions = []string{revision}
 	}
 
-	compareResult := m.CompareAppState(app, proj, revisions, sources, false, true, syncOp.Manifests, app.Spec.HasMultipleSources())
+	// error does not matter as noRevisionCache is set to true
+	compareResult, _ := m.CompareAppState(app, proj, revisions, sources, false, true, syncOp.Manifests, app.Spec.HasMultipleSources())
```

Comment:
Maybe should go ahead and handle the error, ignoring only if it's the ignorable type of error? We know CompareAppState only returns one type of error now, but if another is introduced later, we will probably forget to handle it here.

Reply:
Updated